ScarletEngine/Sources/AchillesPhysics/Private/ColliderSystem.cpp

```diff
@@ -0,0 +1,114 @@
+#include "ColliderSystem.h"
+#include "RigidBodyComponent.h"
+#include "Core.h"
+
+namespace ScarletEngine::Achilles
+{
+	void AABBvsAABBColliderSystem::Update() const
+	{
+	}
+
+	void AABBvsAABBColliderSystem::FixedUpdate() const
+	{
+		// Move collider
+		for (const auto& [Entity, Box, Rb] : GetEntities<BoxColliderComponent, RigidBodyComponent>())
+		{
+			glm::vec3 Dx = Rb->Velocity * (float)FIXED_UPDATE_S;
+			Box->Max += Dx;
+			Box->Min += Dx;
+		}
+
+		// @todo : bounce
+	}
+
+	void SphereVsSphereColliderSystem::Update() const
+	{
+	}
+
+	void SphereVsSphereColliderSystem::FixedUpdate() const
+	{
+		// Move collider
+		for (const auto& [Entity, Rb, Sphere] : GetEntities<RigidBodyComponent, SphereColliderComponent>())
+		{
+			Sphere->Pos += Rb->Velocity * (float)FIXED_UPDATE_S;
+		}
+
+		// Compute intersection
+		const auto& Entities = GetEntities<Transform, SphereColliderComponent, RigidBodyComponent>();
+		int size = Entities.size();
+		for (int i = 0; i < size - 1; i++)
+		{
+			const auto& [EntityA, TransA, SphereA, RbA] = Entities.at(i);
+			for (int j = i + 1; j < size; j++)
+			{
+				const auto& [EntityB, TransB, SphereB, RbB] = Entities.at(j);
+				std::pair<glm::vec3, float> IntersectionDepth = GetIntersection(SphereA, SphereB);
+
+				if (IntersectionDepth.second < 0.f)
+				{
+					glm::vec3 Fi = - (glm::length(RbA->Velocity - RbB->Velocity) * IntersectionDepth.first) / ((float)FIXED_UPDATE_S * (1 / RbA->Mass + 1 / RbB->Mass));
+
+					SolveIntersection(RbA, TransA, SphereA, SphereB, Fi, IntersectionDepth.first * IntersectionDepth.second);
+					SolveIntersection(RbB, TransB, SphereB, SphereA, -Fi, -IntersectionDepth.first * IntersectionDepth.second);
+				}
+			}
+		}
+	}
+
+	std::pair<glm::vec3, float> SphereVsSphereColliderSystem::GetIntersection(SphereColliderComponent* SphereA, SphereColliderComponent* SphereB) const
```

Comment:
What is this `std::pair<glm::vec3, float>` type used everywhere?

Maybe you can refactor this into a struct or at least give it an aliased name so its more clear to the reader what type of data is being returned.